snap/info_snap_yaml_test.go

```diff
@@ -1336,21 +1336,65 @@ apps:
    post-stop-command: post-stop-cmd
    restart-condition: on-abnormal
    bus-name: busName
+   sockets:
+     sock1:
+       listen-stream: /tmp/sock1.socket
+       socket-mode: 0666
 `)
 	info, err := snap.InfoFromSnapYaml(y)
 	c.Assert(err, IsNil)
+
+	app := snap.AppInfo{
+		Snap:            info,
+		Name:            "svc",
+		Command:         "svc1",
+		Daemon:          "forking",
+		RestartCond:     snap.RestartOnAbnormal,
+		StopTimeout:     timeout.Timeout(25 * time.Second),
+		StopCommand:     "stop-cmd",
+		PostStopCommand: "post-stop-cmd",
+		BusName:         "busName",
+		Sockets:         map[string]*snap.SocketInfo{},
+	}
+
+	app.Sockets["sock1"] = &snap.SocketInfo{
+		App:          &app,
+		Name:         "sock1",
+		ListenStream: "/tmp/sock1.socket",
+		SocketMode:   0666,
+	}
+
+	c.Check(info.Apps, DeepEquals, map[string]*snap.AppInfo{"svc": &app})
+}
+
+func (s *YamlSuite) TestDaemonListenStreamAsInteger(c *C) {
+	y := []byte(`name: wat
+version: 42
+apps:
+ svc:
+   command: svc
+   sockets:
+     sock:
+       listen-stream: 8080
```

Comment:
Would it be worth validating that, in such a case as this, the app has at least the `network-bind` socket? Otherwise we know right here that it will fail, right?

Reply:
This is indeed required and validated, see https://github.com/snapcore/snapd/pull/3916/commits/8d9181494282dc47758062d602011fea0925be9b

Reply:
You're on top of things @albertodonato!